extensions/wikia/SeoCrossLink/SeoCrossLinkHelper.class.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * Class SeoCrossLinkHelper
+ */
+class SeoCrossLinkHelper extends WikiaModel {
+
+	const CACHE_TTL = 86400;
+
+	/**
+	 * Show the module only on article pages (for Anon users only)
+	 * @return boolean
+	 */
+	public function canShowModule() {
+		if ( $this->wg->User->isLoggedIn() || !$this->app->checkSkin( 'oasis' ) ) {
+			return false;
+		}
+
+		$title = $this->wg->Title;
+		if ( $title instanceof Title && $title->getNamespace() == NS_MAIN ) {
+			$validPages = $this->getValidPages();
+			if ( in_array( $title->getArticleID(), $validPages ) ) {
+				return true;
+			}
+			return false;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get list of pages that the module is enabled
+	 * @return array
+	 */
+	protected function getValidPages() {
+		$cacheKey = wfMemcKey( 'seo_crosslink', 'valid_pages' );
+		$db = wfGetDB( DB_SLAVE, [], 'specials' );
+		$pages = (new WikiaSQL())->cache( self::CACHE_TTL, $cacheKey, true )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)